source/constructs/api/search/main.py

```diff
@@ -118,3 +126,36 @@ def tables():
     for searchable_class in searchable:
         tables.append(searchable_class.__tablename__)
     return tables
+
+
+@router.get("/download-logs", response_class=StreamingResponse)
+def download_log_as_zip():
+    filename = f"aws_sdps_cloudwatch_logs.zip"
+    logs = boto3.client('logs')
+    response = logs.describe_log_groups(logGroupNamePattern='APIAPIFunction')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `describe_log_groups` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)